src/api/cvc4cpp.cpp

```diff
@@ -5678,6 +5678,74 @@ SmtEngine* Solver::getSmtEngine(void) const { return d_smtEngine.get(); }
  */
 Options& Solver::getOptions(void) { return d_smtEngine->getOptions(); }
 
+/* -------------------------------------------------------------------------- */
+/* Optimization                                                               */
+/* -------------------------------------------------------------------------- */
+
+Objective:: Objective(Term t, ObjectiveType type){
+  d_type = type;
```

Comment:
I think typically we instantiate with instantiation lists rather than in the body of the constructor. But please check with other constructors.